examples/shopping/cart.flow.yaml

```diff
@@ -0,0 +1,279 @@
+import:
+  - products.flow.yaml
+
+collections:
+  - name: examples/shopping/updateCarts
+    schema: schema.yaml#/$defs/updateCart
+    key: [/userId, /productId]
+
+  - name: examples/shopping/purchaseCarts
+    schema: schema.yaml#/$defs/purchaseCart
+    key: [/userId]
+
+  - name: examples/shopping/updateCartsWithProducts
+    key: [/action/userId, /product/id]
+    schema:
+      type: object
+      properties:
+        action: { $ref: schema.yaml#/$defs/updateCart }
+        product: { $ref: schema.yaml#/$defs/product }
+      required: [action, product]
+      reduce: { strategy: lastWriteWins }
+
+    derivation:
```

Comment:
some comments in here would go a long way. i'm having trouble following it myself.

Consider that plenty of people will want to simply flip through the yaml without reading the docs. Some terse pointers would be appreciated.

Consider also breaking this up into multiple flow.yamls, one for each complex derivation, and possibly even with tests in another file.